Get rid of FEATURE_COMMON_PAL and FEATURE_NANOSTACK

SeppoTakalo opened the pull request:
### Description
Get rid of `FEATURE_COMMON_PAL` and `FEATURE_NANOSTACK`
Nanostack related files moved under 'feature/nanostack'
Common libraries moved to 'features/frameworks'

Allow FEATURE_COMMON_PAL and FEATURE_NANOSTACK still to be defined in the build so
that we don't break any builds.

This change is required for future when we start enabling network related tests that depend on the common libraries that are otherwise moved to `FEATURE_COMMON_PAL` or `FEATURE_NANOSTACK` which in first place was temporary solution. None of those libraries under these two folders are target specific.

### Pull request type

    [ ] Fix
    [X] Refactor
    [ ] New target
    [ ] Feature
    [ ] Breaking change

### Effects on RAM/ROM usage
RAM: none
ROM: depend on the linkker.

Tested mbed-os-example-blinky with GCC 6.3.1 20170620 and Flash hit was less than 1kB.

#### master branch:

```
Link: mbed-os-example-blinky
Elf2Bin: mbed-os-example-blinky
+------------------+-------+-------+------+
| Module           | .text | .data | .bss |
+------------------+-------+-------+------+
| [fill]           |   126 |     4 | 2573 |
| [lib]/c.a        | 22945 |  2472 |   89 |
| [lib]/gcc.a      |  3144 |     0 |    0 |
| [lib]/misc       |   248 |     8 |   28 |
| [lib]/nosys.a    |    32 |     0 |    0 |
| main.o           |    56 |     0 |    4 |
| mbed-os/drivers  |    46 |     0 |    0 |
| mbed-os/features |    42 |     0 |  184 |
| mbed-os/hal      |  1611 |     4 |   68 |
| mbed-os/platform |  1913 |   260 |   21 |
| mbed-os/rtos     | 10095 |   168 | 6073 |
| mbed-os/targets  |  8127 |    12 |  384 |
| Subtotals        | 48385 |  2928 | 9424 |
+------------------+-------+-------+------+
Total Static RAM memory (data + bss): 12352 bytes
Total Flash memory (text + data): 51313 bytes
```

#### After this change

```
+------------------+-------+-------+------+
| Module           | .text | .data | .bss |
+------------------+-------+-------+------+
| [fill]           |   135 |     4 | 2477 |
| [lib]/c.a        | 22945 |  2472 |   89 |
| [lib]/gcc.a      |  3144 |     0 |    0 |
| [lib]/misc       |   248 |     8 |   28 |
| [lib]/nosys.a    |    32 |     0 |    0 |
| main.o           |    56 |     0 |    4 |
| mbed-os/drivers  |   657 |     0 |    0 |
| mbed-os/features |   142 |     0 |  280 |
| mbed-os/hal      |  1783 |     4 |   68 |
| mbed-os/platform |  1925 |   260 |   21 |
| mbed-os/rtos     | 10095 |   168 | 6073 |
| mbed-os/targets  |  8127 |    12 |  384 |
| Subtotals        | 49289 |  2928 | 9424 |
+------------------+-------+-------+------+
Total Static RAM memory (data + bss): 12352 bytes
Total Flash memory (text + data): 52217 bytes
```

### Future

After this change there will be

* FEATURE_BLE
* FEATURE_LWIP
* FEATURE_UVISOR

And after EMAC work, the `FEATURE_LWIP` can go as well as it becomes driver independent, and all drivers became stack independent.

Commit:
Get rid of FEATURE_COMMON_PAL

Nanostack related files moved under 'feature/nanostack'
Common libraries moved to 'features/frameworks'

Allow FEATURE_COMMON_PAL still to be defined in the build so
that we don't break any builds.

Commit:
Get rid of FEATURE_NANOSTACK

Allow FEATURE_NANOSTACK still to be defined in the build so
that we don't break any builds.

OPpuolitaival reviewed (approved):
Looks good and we really need this but I am not right person to look at code level

kjbracey reviewed (dismissed):
Approve - only caveat is that in the past I've seen `git subtree pull` refuse to work when the destination has moved, so that might cause problems updating some of the components in future.

If that's still the case, I believe @mikaleppanen should still have a copy of a tiny tweak to the `git-subtree.sh` script to avoid that.

Commit:
Fix Doxygen paths

After moving FEATURE_COMMON_PAL and FEATURE_NANOSTACK Doygen
option file needs to be fixed as well.

sg- reviewed (dismissed):
LGTM 👍

Commit:
Move EFR32 RF driver to TARGET_Silicon_Labs/TARGET_SL_RAIL

This folder structure is identical to where the target code
is found from targets directory.

geky committed:
Added nanostack to removed directories during rtos-less build
